src/bindings/jna/libelektra/src/main/java/org/libelektra/KDB.java

```diff
@@ -271,6 +219,7 @@ private KDB (Pointer pointer)
 	 *                                  released
 	 * @throws IllegalArgumentException if {@code keySet} or {@code parentKey} is
 	 *                                  {@code null}
+	 * @see #get(Key)
 	 */
 	public KDB get (KeySet keySet, Key parentKey) throws KDBException
```

Comment:
The intend for providing both signatures (`KeySet get(Key)` and `KDB get(KeySet, Key)`) is to preserve the ability for re-using an existing key set while also relieving the caller of the need to allocate a target key set beforehand calling `get`. In Java it is common practice to provide a fluent interface whenever possible.
In this case the fluent interface enables chaining multiple `KDB get(KeySet, Key)` and `KDB set(KeySet, Key)` calls. I would argue, that it is better to support this use case than not supporting it, because i don't see any drawbacks resulting from supporting it.
Ultimately it is a design decision: Do we want to provide a fluent API where applicable?.

Reply:
I get why `get(KeySet, Key)` returns `KDB` and not `void`. But I don't really get why I don't think there is much use for a fluent API here.

Chaining like this `kdb.set (ks, parent).get(ks, parent)` could make sense, but I don't see why I would want to call `get` and then immediately `set`. Unless there is something broken somewhere the `set` call would never do anything, because `ks` wasn't modified since `get`.

Reply:
A use case coming to mind would be loading more than one key set for different parent keys into the same or different keysets: `kdb.get(ks, pk1).get(ks, pk2)`

Even if that use case may not come up that frequent, i don't see it hurting the API in any way, like enabling or promoting to misuse it.

Again however we decide, it would definitely make sense to establish a rule when to enable a fluent API and when not to. I would propose to provide a fluent API if here is at least one valid use-case.

Reply:
I think the double `get` example could be pretty confusing, but I guess returning `KDB` doesn't hurt and you can still ignore it and use the function as if it returned `void`. So let's just keep the `KDB` return type.

Reply:
if there are no further objections (@markus2330 you gave a thumbs-up to the initial comment), i would mark this conversation as resolved...